### PR TITLE
Mark operator<< noexcept

### LINKED ISSUE #930: Mark all operator<< noexcept
# Bug reports:

Not all `operator<<` are marked noexcept but should be

### PR DESCRIPTION
**Pull request by fmahebert**
## Proposed changes

Fixes #930. I've split the changes into a few commits, but they could be squashed if preferred.

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by nilsdeppe — DISMISSED** [review]

**Comment by fmahebert** [comment]
@wthrowe or @nilsdeppe: thoughts on the clang-tidy complaint?

**Comment by nilsdeppe** [comment]
I've never understood it or exactly how to fix it without breaking a bunch of other code. In the past we ignored it and since it's unrelated to this PR I'd say we can continue to :)

File: src/Domain/BlockId.hpp [hunk 1]
```diff
@@ -53,7 +53,6 @@ class BlockId {
   }
 
  private:
-
```
Comment (fmahebert) [inline src/Domain/BlockId.hpp:56]: ... this commit message should say "clang-format" not "clang-tidy".

Comment (fmahebert) [reply src/Domain/BlockId.hpp:56]: self rekt

**Comment by fmahebert** [comment]
I've fixed the "clang-format" commit message error, and squashed the multiple "operator<<" commits into one.

**Review by moxcodes — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsdeppe** [comment]
Ignoring the weird clang-tidy error:
```
/work/spectre/src/Parallel/Printf.hpp:75:27: warning: do not implicitly decay an array into a pointer; consider using gsl::array_view or an explicit cast instead [cppcoreguidelines-pro-bounds-array-to-pointer-decay]
  printer(format.c_str(), get_printable_type(std::forward<Ts>(t))...);
                          ^
```

**Pull request merged**